typed-racket-lib/typed-racket/types/union.rkt

```diff
@@ -30,6 +30,19 @@
     (cond
       [(subtype t* t) (list t)]
       [(subtype t t*) ts]
+      ;; the union of two box types is a box type where the write type has to
+      ;; satisfy both write types, and the read type can satisfy either of the
+      ;; two read types
+      [(and (Box? t) (ormap Box? ts))
+       (match* (t ts)
+         [((Box: a-w a-r) (list-no-order (Box: b-w b-r) bs ...))
+          (define w
+            ;; should this use some sort of intersection, or would that
```

Comment:
Can you use `intersect` from `infer/intersect.rkt` here?

I think that would be better than the `cond`

Reply:
I'm not sure what the implications of using `intersect` might be. @pnwamk what do you think?

Reply:
@pnwamk In what situations is it appropriate to use `intersect`?

Reply:
Hey sorry -- I'm a little slow coming up to speed on this.

`intersect` is supposed to compute the logical "and" of two types (performing some simplifications where possible), so if something is both of type `T1` and type `T2`, then saying it is of type `(intersect T1 T2)` should always be fine.

I guess the only "exception" I can think of is in a few places our handling of intersections is not complete (i.e. at the moment we do not fully handle the application of an intersection of function types -- but this isn't a problem with `intersect` or intersection types, it's just a deficiency in our code that handles function application at the moment).

Anyway, did that answer your question?

Reply:
So I'm guessing it is appropriate to use `intersect` in `normalize-type` to determine the "write" type of the merged result of a union of two box types?

Reply:
(Re: https://github.com/racket/typed-racket/pull/225#issuecomment-354801561)
Okay, I've changed this to use `intersect`.